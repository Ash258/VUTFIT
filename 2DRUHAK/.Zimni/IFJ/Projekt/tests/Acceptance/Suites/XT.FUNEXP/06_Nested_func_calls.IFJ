' Mested function calls with multiple data types

function blank() as integer
    print !"blank";
    return 42 * 10 / 8
end function

function dbl(par as double, par2 as integer) as double
    print par;
    print !"dbl";
    print par2;
    return 2.78 / par + par2 * 0.3
end function

function int_(par as integer) as string
    print !"int";
    print par;
    return !"barman"
end function

function str_(par as string) as double
    print !"str";
    print par;
    return 75 * 78.74 / 0.12345
end function

scope
    dim var_ as integer = str_(int_(dbl(blank(), blank())))
    print var_;
end scope
