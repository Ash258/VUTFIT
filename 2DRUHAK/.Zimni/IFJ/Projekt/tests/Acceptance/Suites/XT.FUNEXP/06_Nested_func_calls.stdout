blankblank 52dbl 52int 16strbarman 47837